src/lib/modules/charts/components/dataset-toggle/dataset-toggle.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+import { City, ClimateModel, Dataset, DatasetService } from '../../../../public_api';
```

Comment:
I'm not a huge fan of importing internal modules from `public_api` like this, for several reasons:
 - It conflates the external visibility of packages with their internal visibility (I may want to share a module amongst modules in  `climate-change-components` but not share it with consumers of the library).
 - It negates/discourages placing modules in the appropriate location, since they're all just being imported in a grab-bag fashion regardless.

_Consider_ switching all imports from `public_api` to imports from the module's actual location.

Reply:
Great point. I did the first import like that, and just kept going for consistency, and figured I'd change it later in one go if it became a problem which obviously didn't happen. 

It's a bit unclear from your comment, which of the following were you referring to?

### (1)
```
import { City } from '../../../api/models/city.model';
...
```
or

### (2)
```
import { City, ... } from '../../../api';
```

Reply:
IMHO (1) is preferable to (2) which is preferable to the current import from `public_api`

Reply:
Agree. (2) just pushes the same problems up a layer, so just wanted to be clear. Will update with (1).

Reply:
I should add here that the api/chart/shared module index.ts files do a mix of `*` and explicit imports. I think I'd prefer that these files always do explicit imports, for the same reasoning that we prefer explict imports in Python. It's easier to see what the public API surface is at a glance in one place and it avoids cluttering namespaces or accidentally exposing something that shouldn't be.

Thoughts on cleaning those up as well, since I've already done some significant QoL refactors?

Reply:
Go for it :clap: